### ISSUE TITLE
Specifying more than one `match_expression`  in the `kubernetes_resource_quota` results in a crash

### ISSUE THREAD

**Issue by anatolek** [hashicorp/terraform-provider-kubernetes#1561]
### Terraform Version, Provider Version and Kubernetes Version
```
Terraform version: v1.1.2
Kubernetes provider version: v2.7.1
Kubernetes version: v1.21.2-eks-06eac09
```

### Affected Resource(s)
- kubernetes_resource_quota

### Terraform Configuration Files
```hcl
resource "kubernetes_resource_quota" "example" {
  metadata {
    labels = {
      "Owner"     = "DevOps"
      "Terraform" = "true"
    }
    name      = "test-tf-ns-exemple"
    namespace = "test-tf-ns"
  }
  spec {
    hard = {
      pods = 10
    }
    scopes = ["BestEffort"]
    scope_selector {
      match_expression {
        operator = "In"
        scope_name = "PriorityClass"
        values = ["medium"]
      }
      match_expression {
        operator = "In"
        scope_name = "PriorityClass"
        values = ["low"]
      }
    }
  }
}
```

### Panic Output
https://gist.github.com/anatolek/09445b8fb611e4a6def5a8914f3f2f54

### Steps to Reproduce
1. Create k8s namespace `test-tf-ns`
1. Create a module that includes the `kubernetes_resource_quota` resource with 2 `match_expression` blocks
1. `terraform apply` 

### Expected Behavior
Terraform should create a resource without issues

### Actual Behavior
Terraform created the resource in Kubernetes but did not write it into the state and crashed with an error

### Important Factoids
- I use AWS EKS cluster

### References
- none

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by arybolovlev** [comment]
Hi @anatolek,

Thank you for reporting this issue. The fix is ready([PR](https://github.com/hashicorp/terraform-provider-kubernetes/pull/1658)) and waiting for review. I will attach PR to this issue and it will be closed as soon as the fix is merged. Once it is merged into the main branch, it will be available in the upcoming release. Please keep an eye on release notes.

Stay safe!
